zulipterminal/helper.py

```diff
@@ -490,6 +490,11 @@ def match_emoji(emoji: str, text: str) -> bool:
     return emoji.lower().startswith(text.lower())
 
 
+def match_stream_names(stream_names: List[str], search_text: str) -> List[str]:
+    return [name for name in stream_names
+            if name.lower().startswith(search_text.lower())]
+
+
```

Comment:
There are no pinned stream considerations here - I'm not using the existing function.
Refactoring `match_streams` looks a little hard, and pinning order is not super important - so I'd like to deal with this in a follow-up.

Reply:
Though one hacky way to use the current `match_streams` would be to pass `[(stream_name, stream_name)..]` in data`.

Reply:
They are certainly an improvement, but I don't see it as being a necessity for this particular feature. I'll make a temp fix but open an issue and make a follow-up PR for refactoring `match_stream`